app/src/org/commcare/dalvik/application/CommCareApplication.java

```diff
@@ -908,6 +906,7 @@ public void onServiceDisconnected(ComponentName className) {
         // class name because we want a specific service implementation that
         // we know will be running in our own process (and thus won't be
         // supporting component replacement by other applications).
+        startService(new Intent(this, CommCareSessionService.class));
         bindService(new Intent(this, CommCareSessionService.class), mConnection, Context.BIND_AUTO_CREATE);

```

Comment:
Binding the service with an auto-start flag doesn't ensure that `onTaskRemoved` is called when the app is swipe-closed. Hence we have to manually call `startService` and then bind it.